Lib/test/test_xpickle.py

```diff
@@ -0,0 +1,260 @@
+# This test covers backwards compatibility with
+# previous version of Python by bouncing pickled objects through Python 3.6
+# and Python 3.9 by running xpickle_worker.py.
+import os
+import pathlib
+import pickle
+import subprocess
+import sys
+
+
+from test import support
+from test import pickletester
+from test.test_pickle import PyPicklerTests
+
+try:
+    import _pickle
+    has_c_implementation = True
+except ModuleNotFoundError:
+    has_c_implementation = False
+
+is_windows = sys.platform.startswith('win')
+
+# Map python version to a tuple containing the name of a corresponding valid
+# Python binary to execute and its arguments.
+py_executable_map = {}
+
+def highest_proto_for_py_version(py_version):
+    """Finds the highest supported pickle protocol for a given Python version.
+    Args:
+        py_version: a 2-tuple of the major, minor version. Eg. Python 3.7 would
+                    be (3, 7)
+    Returns:
+        int for the highest supported pickle protocol
+    """
+    major = sys.version_info.major
+    minor = sys.version_info.minor
+    # Versions older than py 3 only supported up until protocol 2.
+    if py_version < (3, 0):
+        return 2
+    elif py_version < (3, 4):
+        return 3
+    elif py_version < (3, 8):
+        return 4
+    elif py_version <= (major, minor):
+        return 5
+    else:
+        # Safe option.
+        return 2
```

Comment:
I agree it looks strange. However, I considered the case where once future versions of python are released and the test users run the test with an older version of python, that might occur. Eg a user runs the test for python 3.11 using python 3.10. 

It might be better to return protocol 5 instead though.